ipcore/hw_tcl_generator.py

```diff
@@ -0,0 +1,268 @@
+
+
+class HwTCLGenerator:
+    def __init__(self, dataplane_config):
+        self.name = dataplane_config.name
+        self.version = dataplane_config.version
+        self.description = dataplane_config.description
+        self.author = dataplane_config.author
+
+        self.additional_filesets = dataplane_config.additional_filesets
+
+        self.vendor = dataplane_config.vendor
+
+        self.clock_rate = dataplane_config.clock_rate
+
+        self.data_bus_size = dataplane_config.data_bus_size
+        self.address_bus_size = dataplane_config.address_bus_size
+        self.has_avalon_mm_slave_signal = dataplane_config.has_avalon_mm_slave_signal
+
+        self.sink_max_channel = dataplane_config.sink_max_channel
+        self.source_max_channel = dataplane_config.source_max_channel
+
+    def write_tcl(self, outputFilename):
+        with open(outputFilename, "w") as out_file:
+            out_file.write(self.generate())
+    
+    def generate(self):
+        tcl = self.create_header_file_stuff()
+        tcl += self.create_module()
+        tcl += self.create_file_sets()
+        tcl += self.create_module_assignments()
+        tcl += self.create_connection_point_clock()
+        tcl += self.create_connection_point_reset()
+        tcl += self.create_mm_connection_point()
+        tcl += self.create_sink_connection_point()
+        tcl += self.create_source_connection_point()
+        return tcl
+
+    def create_header_file_stuff(self):
+        tcl = "# # # # # # # # # # # # # # # # # #\n"
+        tcl += "# Built in create_header_file_stuff\n"
```

Comment:
Should we remove these comment headers from the generated file? I'm not convinced that they are useful, and the formatting between comment headers does not seem to be consistent right now.

Reply:
Sure